pkg/controller/sparkapplication/sparkapp_util.go

```diff
@@ -38,7 +38,11 @@ func getSparkApplicationID(pod *apiv1.Pod) string {
 	return pod.Labels[config.SparkApplicationSelectorLabel]
 }
 
-func getDefaultDriverPodName(app *v1beta1.SparkApplication) string {
+func getDriverPodName(app *v1beta1.SparkApplication) string {
```

Comment:
Since driver pod name can be set in sparkConf as well, we probably should check that as well here before defaulting. I came up with something like https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/495/files#diff-d110a4ca56d08f6d4992d143e985ee66R41

Reply:
Done.